src/engine_core/src/utils/StringUtils.cpp

```diff
@@ -0,0 +1,15 @@
+#include "StringUtils.hpp"
+
+#if _WIN32
+using UTF8ToUTF16Convert = std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>>;
+std::wstring StringUtils::ToWString(const char *data)
+{
+    UTF8ToUTF16Convert converter;
+    std::wstring convertedStr = converter.from_bytes(data);
+    return convertedStr;
+}
+std::string StringUtils::FromWString(std::wstring str)
+{
+    return std::string(str.begin(), str.end());
```

Comment:
This is not the best thing to do, some compilers generate a warning as both have a different size, so some data might get lost.

Reply:
I'll see if I can get rid of this with another approach